src/nvim/os/shell.c

```diff
@@ -640,24 +632,19 @@ static size_t write_output(char *output, size_t remaining, bool to_buffer,
 
     if (output[off] == NUL) {
       // Translate NUL to NL
-      output[off] = replacement_NUL;
+      output[off] = NL;
     }
     off++;
   }
 
   if (eof) {
     if (remaining) {
-      if (to_buffer) {
-        // append unfinished line
-        ml_append(curwin->w_cursor.lnum++, (char_u *)output, 0, false);
-        // remember that the NL was missing
-        curbuf->b_no_eol_lnum = curwin->w_cursor.lnum;
-      } else {
-        screen_del_lines(0, 0, 1, (int)Rows, NULL);
-        screen_puts_len((char_u *)output, (int)remaining, lastrow, 0, 0);
-      }
+      // append unfinished line
+      ml_append(curwin->w_cursor.lnum++, (char_u *)output, 0, false);
+      // remember that the NL was missing
+      curbuf->b_no_eol_lnum = curwin->w_cursor.lnum;
       output += remaining;
-    } else if (to_buffer) {
+    } else {
       curbuf->b_no_eol_lnum = 0;
```

Comment:
the other changes treat `to_buffer` as always `false`, so maybe this should be removed.